extensions/chghost-3.2.md

```diff
@@ -16,20 +16,37 @@ The chghost client capability allows servers to directly inform clients about a
 host or user change without having to send fake quits or joins. This capability
 MUST be referred to as `chghost` at capability negotiation time.
 
-When enabled, clients will get the `CHGHOST` message to designate the host of a
-user changing for clients on common channels with them.
+When the capability is enabled, servers MUST send the `CHGHOST` message to
+clients sharing the same channel as the target client, informing those clients
+that the user or host of the target client has changed.
+
+Servers SHOULD also send the `CHGHOST` command to the client who changed their
+host, as well as the previously-unspecified `396` numeric, whose exact value
+count changes per implementation (from two to three arguments), but always
+places the client's new host in the second-to-last position.
 
 The `CHGHOST` message is as follows:
 
-    :nick!user@host CHGHOST new-user new.host.goes.here
+    :nick!user@old_host.local CHGHOST new-user new_host.local
+
+The `new-user` parameter represents the user's "username" or "ident" which may
+or may not have changed in the CHGHOST process.
+
+The `new_host.local` parameter represents the new hostname for the user which
+may or may not have changed in the CHGHOST process.
 
-The field represented by `new-user` represents the user's "username" or "ident"
-which may or may not have changed in the CHGHOST process.
+When the capability is not enabled for clients who share the same channel,
+servers should fall back to having the client `QUIT` the server, rejoin all
```

Comment:
It should be made extremely clear that this is a **fake** leaving and rejoining the channel. May be worth discussing whether recommending `PART` is nicer as the server won't also need to do fake MONITOR events and such? I wonder how existing servers handle this and whether this behaviour matches that in-the-wild behaviour.

Reply:
I was going to bring it up - and thought I already had - but I was thinking that it should act as a full disconnect from the client in a way that MONITOR detects them going offline and coming back online, as that would probably be the only way to send the information. I'm open to discussing whether or not it should act as a `/join 0` or a full disconnect and reconnect.

Reply:
Important to note that existing implementations (Freenode at the very least) treat it like a full disconnect at the very least for the channels that the user is in - treating it like a QUIT rather than a PART.

Reply:
> Upon a client changing their details with the `USERHOST` command, other clients sharing a channel with that receive the updated username and hostname. Clients that have the capability negotiated receive a `USERHOST` message indicating the new username and hostname of the given client. Clients that do not have the capability negotiated MUST see the given client `QUIT` and proceed to reconnect with the new username and hostname.

I believe this explains how clients should act nicely. It explains both cases in one place rather than at either end of the document.

I'd replace the _"When the capability is enabled, servers MUST"_ paragraph near the top with this rewritten one here. After that, I'd remove this _"When the capability is not enabled"_ paragraph entirely.